src/Microsoft.Diagnostics.ExtensionCommands/DumpHttpCommand.cs

```diff
@@ -0,0 +1,298 @@
+
+                foreach (KeyValuePair<HttpRequestStatGroupKey, HttpRequestStat> entry in statCounts.OrderByDescending(s => s.Value.Count))
+                {
+                    output.WriteRow(entry.Value.Count, entry.Value.HttpMethod, entry.Value.StatusCode, entry.Value.Host);
+                }
+
+                int total = statCounts.Select(stat => stat.Value.Count).Sum();
+                WriteLine($"Total {total} requests");
+            }
+
+            // <summary>Render each http request.</summary>
+            void RenderRequests()
+            {
+                Table output = new(Console, ColumnKind.Pointer, ColumnKind.Pointer, s_httpMethodColumn, s_statusCodeColumn, ColumnKind.Text);
+                output.WriteHeader("Address", "MethodTable", "Method", "StatusCode", "Uri");
+
+                foreach (HttpRequestInfo httpRequest in httpRequests)
+                {
+                    output.WriteRow(httpRequest.Address, httpRequest.MethodTable, httpRequest.HttpMethod, httpRequest.StatusCode, httpRequest.Url);
+                }
+            }
+        }
+
+        private IEnumerable<HttpRequestInfo> CollectHttpRequests()
+        {
+            IEnumerable<ClrObject> objectsToPrint = FilteredHeap!.EnumerateFilteredObjects(Console.CancellationToken);
+
+            foreach (ClrObject clrObject in objectsToPrint)
+            {
+                if (clrObject.Type?.Name == "System.Net.Http.HttpRequestMessage")
+                {
+                    // HttpRequestMessage doesn't have reference to HttpResponseMessage
+                    // the same http request can be found by HttpResponseMessage
+                    // these duplicates handled by FilterDuplicates method
+                    yield return BuildRequest(clrObject, null);
+                }
+
+                if (clrObject.Type?.Name == "System.Net.Http.HttpResponseMessage")
+                {
+                    ClrObject request = GetRequest(clrObject);
+                    yield return BuildRequest(request, clrObject);
+                }
+
+                // TODO handle System.Net.HttpWebRequest for .NET Framework dumps
+            }
+
+            yield break;
+
+            HttpRequestInfo BuildRequest(ClrObject request, ClrObject? response)
+            {
+                string httpMethod = GetHttpMethod(request);
+                string uri = GetUri(request);
+                int? statusCode = response != null ? GetStatusCode(response.Value) : null;
+
+                return new HttpRequestInfo(
+                    request.Address,
+                    request.Type!.MethodTable,
+                    httpMethod,
+                    uri, statusCode);
+            }
+
+            string GetHttpMethod(ClrObject request)
+            {
+                return Runtime.ClrInfo.Flavor == ClrFlavor.Core
+                    ? request.ReadObjectField("_method").ReadStringField("_method")!
```

Comment:
In the case of field name change, will the exception bubble to the user in a way that makes sense?

Reply:
Current error message: `Type 'System.Net.Http.HttpRequestMessage' does not contain a field named '_method'`. I can add additional message like 'Unable to read HTTP method from request'. Error message text proposal is welcome.

Reply:
Changed error message to `Unable to read HTTP Method. Type 'System.Net.Http.HttpRequestMessage' does not contain any field named _method or method`